modules/secret-manager/main.tf

```
@@ -0,0 +1,51 @@
+data "aws_iam_policy_document" "read_policy" {
+  statement {
+    actions = [
+      "secretsmanager:ListSecrets"
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "secretsmanager:DescribeSecret",
+      "secretsmanager:GetSecretValue",
+    ]
+
+    resources = var.secrets
+  }
+}
+
+data "aws_iam_policy_document" "write_policy" {
+  statement {
+    actions = [
+      "secretsmanager:ListSecretVersionIds",
+      "secretsmanager:PutSecretValue",
+    ]
+
+    resources = var.secrets
+  }
+}
+
+resource "aws_iam_policy" "read_policy" {
+  name   = "read_policy"
+  policy = data.aws_iam_policy_document.read_policy.json
+}
+
+resource "aws_iam_policy_attachment" "read_attachment" {
```

```suggestion
resource "aws_iam_policy_attachment" "secret_read" {
```
attachment is already in resource name